lms/djangoapps/learner_dashboard/views.py

```diff
@@ -0,0 +1,52 @@
+"""
+Handles requests for views, returning page frame
+"""
+
+import logging
+from urlparse import urljoin
+
+from django.conf import settings
+from django.contrib.auth.decorators import login_required
+from django.views.decorators.http import require_GET
+from django.http import Http404
+
+from edxmako.shortcuts import render_to_response
+from openedx.core.djangoapps.programs.utils import get_programs_for_dashboard
+from learner_dashboard.models import LearnerDashboardConfiguration
+from student.views import get_course_enrollments
+
+
+log = logging.getLogger(__name__)
+
+
+@login_required
+@require_GET
+def view_programs(request):
+    if not LearnerDashboardConfiguration.current().enable_learner_dashboard:
+        raise Http404("learner dashboard not enabled")
+
+    course_enrollments = list(get_course_enrollments(request.user, None, []))
+
+    # Get the programs by courses dictionary from the get_programs_for_dashboard function call
+    programs_dict = get_programs_for_dashboard(
+        request.user,
+        [enrollment.course_id for enrollment in course_enrollments])
+
+    program_list = []
+    # Extract all the programs from the dictionary above
+    for programs_by_courses in programs_dict.values():
+        for program in programs_by_courses:
+            if program not in program_list:
+                # Remove duplicates
+                program['marketing_url'] = urljoin(
+                    settings.MKTG_URLS.get('ROOT'),
+                    'xseries' + '/{}'
+                ).format(program['marketing_slug'])
+                program_list.append(program)
+
+    xseries_url = urljoin(settings.MKTG_URLS.get('ROOT'), 'xseries')
```

Comment:
We should keep open-source site operators in mind here. Someone other than ourselves may want to use this code to list their own programs, but they're not going to be offering XSeries. Can you do this in a way that isn't so closely tied to XSeries? Maybe you can use the config model to store this value?